### ISSUE TITLE
PUID/GUID is being ignored

### ISSUE THREAD

**Issue by hurfi** [JackDallas/Premiumizearr#64]
Since the PUID/GUID problem is resolved with a pre-release build, I think i can close this issue. If @JackDallas  thinks it should be open, please reopen.

_Originally posted by @dabonzo in https://github.com/JackDallas/Premiumizearr/issues/52#issuecomment-1170978777_
      

Has this been pushed to the current docker file?
I'm having the same issue running v1.2.7-rc1 on unraid. Once the download is finished and sent to the finished downloads folder the owner is set to root:root, sonarr and radarr then are denied access to the file.

**Comment by DatPat** [comment]
hello,
I am running a custom version of premiumizarr so I am not sure if this carries over to this version but I can set PUID with "--user 99" as an extra parameter. If this doesn't work you can use https://hub.docker.com/r/datpat/premiumizearr/ which is a version of premiumizearr I made which is extended by the ability to handle torrents.

**Comment by GOvEy1nw** [comment]
@DatPat thank you for this alt version! 
If possible, could you add auth/password protection to your version too? It's the one thing that makes me uneasy about premiumizearr, as all the other *arr's and stuff I use have authentication built in.

**Comment by DatPat** [comment]
Is there a version of premiumizarr which already has authentication? If there isn't, would you prefer a password as a container/env variable or in the html interface?

**Comment by GOvEy1nw** [comment]
I don't believe there is a version with it already. I know that https://github.com/piejanssens/premiumizer does, but it's much slower and quite buggy compared to premiumizearr.

I'm not quite sure on the technical differences between having it as an env variable or html interface, but whichever is easiest to implement? [rdt-client ](https://github.com/rogerfar/rdt-client) (the real-debrid version of premiumizarr) has auth implemented already, so perhaps it could be lifted from that for speed (not sure if that's how it works lol!).

Also, slightly off-topic, but I saw on docker hub that you have an offcloud docker that seems to be similar to this too. Is that still actively worked on also?

**Comment by DatPat** [comment]
I will check out rdt-client and see if I can't just add whatever the author has added to premiumizarr. As for offcloud(brid), yes it's being worked on still but it's a very simple program currently. it basically just scans the blackhole for files and downloads them. Any information is just logged into the logs. But it's my goal to have a premiumizarr-version for offcloud in the future though. I am just not sure when I will have the time to make this a reality, but it's definitly my goal.

**Comment by GOvEy1nw** [comment]
That's great! I use all 3, Premiumize, Real-Debrid and Offcloud, for variance as they all have strengths/weaknesses. So a solution for each would be fantastic. In an ideal world, a solution which combines them all and is then able to see if any of them already have the requested file already in their cache, and then auto download from that one, would be fantastic, at least for my use case.

**Comment by DatPat** [comment]
I checked out RDT and I am pretty impressed with it, thanks for showing it to me. Now I am wondering if the best way forward wouldn't just be to add offcloud support to RDT as well as having the ability to setup multiple providers and use them randomly.

**Comment by GOvEy1nw** [comment]
Yea it has some nice additional features (such as the ability to add new torrents directly into it etc) which I think would be beneficial for all the debrid clients. The only thing that's probably missing is usenet compatibility, because real-debrid doesn't have that, but premiumize (and premiumizearr) and offcloud do.

An all-in-one solution would be fantastic imo, but even if that couldn't work, just the ability to run multiple versions of the same container that each connect to a different debrid type account would be fantastic too, as then there's parity in the features, UI and UX.

**Comment by JackDallas** [comment]
@DatPat if you're maintaining a fork of this and are happy to keep doing so I'd be happy to point at it from the README to get traffic moving towards it.

**Comment by DatPat** [comment]
> @DatPat if you're maintaining a fork of this and are happy to keep doing so I'd be happy to point at it from the README to get traffic moving towards it.

Thanks so much for the offer but I don't think that my project is good enough yet. I only got started recently on golang and any changes I would make could reflect poorly on the original project. I would like to get back to this offer though once I am more confident in my abilities.

**Comment by F43nd1r** [comment]
Looks like @JackDallas is back on this project? If so I'd love to see the changes from @DatPat upstreamed 🙂

**Comment by andreasbrett** [comment]
> hello, I am running a custom version of premiumizarr so I am not sure if this carries over to this version but I can set PUID with "--user 99" as an extra parameter. If this doesn't work you can use https://hub.docker.com/r/datpat/premiumizearr/ which is a version of premiumizearr I made which is extended by the ability to handle torrents.

What do you mean by "--user 99 as an extra parameter"? I tried your docker container but couldn't figure out how to provide this parameter.

**Comment by ghettopato** [comment]
In unraid you can set extra parameters if you tick the advanced switch. If you don't use Unraid, you just have to add the text to the command line somehow.
